Source/Provers/SMTLib/SMTLibBatchTheoremProver.cs

```diff
@@ -97,7 +97,7 @@ public override int FlushAxiomsToTheoremProver()
 
         var result = CheckSat(cancellationToken);
```

Comment:
You're right, it is existing behavior. And it looks weird.
Looking at the control flow, the CheckSat command is supposed to close the input after sending all the commands, synchronously, so the pop will never make it to the process's input. But it will make it to the log.
So I guess it's ok, although I'm not sure why it's necessary.